code/game/objects/structures/displaycase.dm

```diff
@@ -343,3 +362,84 @@
 	name = initial(I.name)
 	icon = initial(I.icon)
 	icon_state = initial(I.icon_state)
+
+/obj/structure/displaycase/forsale
+	name = "vend-a-tray"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "laserbox0"
+	desc = "A display case with an ID-card swipe, to purchase the contents."
+	density = FALSE
+	max_integrity = 100
+	req_access = list(ACCESS_KITCHEN)
+	showpiece_type = /obj/item/reagent_containers/food
+	var/sale_price = 20
+	var/datum/bank_account/payments_acc = null
+
+/obj/structure/displaycase/forsale/update_icon()	//remind me to fix my shitcode later

```

Comment:
out of scope for your PR but someone might be updating the method that /displaycase/update_icon() occurs which may require you to update your PR to follow the same standard.